controllers/articles-controller.js

```diff
@@ -40,4 +40,66 @@ exports.getCommentsForArticle = (req, res, next) => {
     .catch((err) => {
         next(err)
     })
-};
\ No newline at end of file
+};
+
+exports.postComment = (req, res, next) => {
+    const id = req.params.article_id;
+    const body = req.body;
+
+    if (!checkCommentStructure(body)) next({ 
+        statusCode: 400, msg: 'Bad request! Missing or incorrect key name(s)' 
+    });    
+
+    checkUserExists(body.username)
+    .then((userExists) => {
+        if (!userExists) {
+            next({statusCode: 401, msg: 'You need to have an account to post comments' });
+        };
+        return checkArticleExists(id);
+    })
+    .then((articleExists) => {
+        if (!articleExists) {
+            next({ statusCode: 404, msg: 'No article with that ID' });
+        };
+        return insertComment(id, body);    
+    })
+    .then(({rows}) => {
+        res.status(201).send({postedArticle: rows[0]});

```

Comment:
I'd recommend sending it back on a key of just 'article' rather than postedArticle, but just a suggestion.